src/System.Drawing.Common/tests/Imaging/ImageAttributesTests.cs

```diff
@@ -0,0 +1,1503 @@
+        public void SetColorMatrix_ColorMatrix_Success()
+        {
+            using (var brush = new SolidBrush(_actualGreen))
+            using (var bitmap = new Bitmap(_rectangle.Width, _rectangle.Height))
+            using (var graphics = Graphics.FromImage(bitmap))
+            using (var imageAttr = new ImageAttributes())
+            {
+                imageAttr.SetColorMatrix(_colorMatrix);
+                bitmap.SetPixel(0, 0, _actualYellow);
+                graphics.DrawImage(bitmap, _rectangle, _rectangle.X, _rectangle.Y, _rectangle.Width, _rectangle.Height, GraphicsUnit.Pixel, imageAttr);
+                Assert.Equal(_expectedRed, bitmap.GetPixel(0, 0));
+
+                graphics.FillRectangle(brush, _rectangle);
+                graphics.DrawImage(bitmap, _rectangle, _rectangle.X, _rectangle.Y, _rectangle.Width, _rectangle.Height, GraphicsUnit.Pixel, imageAttr);
+                Assert.Equal(_expectedBlack, bitmap.GetPixel(0, 0));
+            }
+        }
+
+        public static IEnumerable<object[]> ColorMatrix_DropShadowRepaintWhenAreaIsSmallerThanTheFilteredElement_TestData()
+        {
+            yield return new object[] { Color.FromArgb(100, 255, 0, 0) };
+            yield return new object[] { Color.FromArgb(255, 255, 155, 155) };
+        }
+
+        [ConditionalTheory(nameof(PlatformDetection) + "." + nameof(PlatformDetection.IsNotWindowsNanoServer))]
+        [MemberData(nameof(ColorMatrix_DropShadowRepaintWhenAreaIsSmallerThanTheFilteredElement_TestData))]
+        public void SetColorMatrix_ColorMatrixI_Success(Color color)
+        {
+            ColorMatrix colorMatrix = new ColorMatrix(new float[][]
+            {
+                new float[] {1, 0, 0, 0, 0},
+                new float[] {0, 1, 0, 0, 0},
+                new float[] {0, 0, 1, 0, 0},
+                new float[] {0, 0, 0, 0.5f, 0},
+                new float[] {0, 0, 0, 0, 1},
+            });
+
+            using (var brush = new SolidBrush(color))
+            using (var bitmapBig = new Bitmap(200, 100))
+            using (var bitmapSmall = new Bitmap(100, 100))
+            using (var graphicsSmallBitmap = Graphics.FromImage(bitmapSmall))
+            using (var graphicsBigBitmap = Graphics.FromImage(bitmapBig))
+            using (var imageAttr = new ImageAttributes())
+            {
+                graphicsSmallBitmap.FillRectangle(Brushes.White, 0, 0, 100, 100);
+                graphicsSmallBitmap.FillEllipse(brush, 0, 0, 100, 100);
+                graphicsBigBitmap.FillRectangle(Brushes.White, 0, 0, 200, 100);
+                imageAttr.SetColorMatrix(colorMatrix);
+                graphicsBigBitmap.DrawImage(bitmapSmall, new Rectangle(0, 0, 100, 100), 0, 0, 100, 100, GraphicsUnit.Pixel, null);
+                graphicsBigBitmap.DrawImage(bitmapSmall, new Rectangle(100, 0, 100, 100), 0, 0, 100, 100, GraphicsUnit.Pixel, imageAttr);
+                Assert.Equal(Color.FromArgb(255, 255, 155, 155), bitmapBig.GetPixel(50, 50));
```

Comment:
I'm a little worried that such specific assertions are going to fail against libgdiplus (unless we have pixel-perfect compatibility, which we won't...), but we can deal with that later.